components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -332,23 +333,23 @@ public void removeExpiredSessionData(Timestamp timestamp) {
         }
         try {
             statement = connection.prepareStatement(sqlDeleteExpiredDataTask);
-            long currentStandardNano = getCurrentStandardNano(timestamp);
-            statement.setLong(1, currentStandardNano);
+            long cleanupLimit = FrameworkUtils.getCurrentStandardNano() - IdentityUtil.getCleanUpTimeout() * 60 * 10^9;
+            statement.setLong(1, cleanupLimit);
             statement.execute();
             if (!connection.getAutoCommit()) {
                 connection.commit();
             }
         } catch (SQLException e) {
-            log.error("Error while removing session data from the database for the timestamp " + timestamp.toString(), e);
+            log.error("Error while removing session data from the database", e);
```

Comment:
It's better to print the `cleanupLimit` so it would have accurate information which task or which iteration it get failed.